ambry-api/src/main/java/com.github.ambry/notification/NotificationSystem.java

```diff
@@ -34,9 +34,10 @@
    * @param account The {@link Account} for the blob
    * @param container The {@link Container} for the blob
    * @param notificationBlobType The {@link NotificationBlobType} of this blob.
+   * @param externalAssetTag The external asset tag of this blob.
    */
   void onBlobCreated(String blobId, BlobProperties blobProperties, Account account, Container container,
```

Comment:
at some point (maybe not this PR), we might want to combine the account, container, notificationBlobType and externalAssetTag arguments into a data structure that gets passed into this method

Reply:
@lightningrob , After a discussion with @cgtz , we can actually create a new class to include `Account`, `Container` and `externalAssetTag`. All non-persistent blob properties can be included in this class in the future. Tentatively name it as `NonPersistentBlobProperties`.

Reply:
Looks like what I proposed above is not a good way. 
Introducing `NonPersistentBlobProperties`/`BlobNotifications` all requires change to ```Future<String> putBlob(BlobProperties blobProperties, byte[] userMetadata, ReadableStreamChannel channel,
      PutBlobOptions options, Callback<String> callback)```. 
I prefer not to change this elegant interface. 
Will add `externalAssetTag` to `BlobProperties` and comment it as non-persistent property.  @cgtz  @lightningrob